pyfstat/mcmc_based_searches.py

```diff
@@ -3322,40 +3318,41 @@ def _initiate_search_object(self):
             self.maxStartTime = self.search.maxStartTime
 
     def logl(self, theta, search):
-        for j, theta_i in enumerate(self.theta_idxs):
-            self.fixed_theta[theta_i] = theta[j]
-        in_theta = copy.copy(self.fixed_theta)
-        in_theta[1] = in_theta[0] + in_theta[1]
-        if in_theta[1] > self.maxStartTime:
+        in_theta = {
+            key: self.fixed_theta[k]
+            for k, key in enumerate(self.full_theta_keys)
+            if "transient" not in key
+        }
+        in_theta.update(
+            {
+                key: theta[k]
+                for k, key in enumerate(self.theta_keys)
+                if "transient" not in key
+            }
+        )
+        # FIXME: this can be simplified when changing all theta lists to dicts
+        if "transient_tstart" in self.theta_keys:
+            in_theta["tstart"] = theta[self.theta_keys.index("transient_tstart")]
+        else:
+            in_theta["tstart"] = self.fixed_theta[
+                self.full_theta_keys.index("transient_tstart")
+            ]
+        if "transient_duration" in self.theta_keys:
+            tau = theta[self.theta_keys.index("transient_duration")]
+        else:
+            tau = self.fixed_theta[self.full_theta_keys.index("transient_duration")]
+        in_theta["tend"] = in_theta["tstart"] + tau
+        if in_theta["tend"] > self.maxStartTime:
             return -np.inf
-        twoF = search.get_fullycoherent_twoF(*in_theta)
+        twoF = search.get_fullycoherent_twoF(**in_theta)
         return twoF / 2.0 + self.likelihoodcoef
 
     def _unpack_input_theta(self):
-        full_theta_keys = [
-            "transient_tstart",
-            "transient_duration",
-            "F0",
-            "F1",
-            "F2",
-            "Alpha",
-            "Delta",
-        ]
+        self.full_theta_keys = ["F0", "F1", "F2", "Alpha", "Delta"]
```

Comment:
I've seen this same list hard coded too many times...

Reply:
Maybe it should be an attribute of the `ComputeFStatistic` core class, `default_doppler_parameter_keys` or something...? But that should be another separate change, I'd say.